gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GitMonitoringService.java

```diff
@@ -426,7 +450,18 @@ protected JSch createDefaultJSch(FS fs) throws JSchException {
           JSch.setLogger(new JschLogger());
         }
         JSch defaultJSch = super.createDefaultJSch(fs);
-        defaultJSch.addIdentity(GitMonitoringService.this.privateKeyPath, GitMonitoringService.this.passphrase);
+        defaultJSch.getIdentityRepository().removeAll();
+        if (GitMonitoringService.this.privateKeyPath != null) {
+          defaultJSch.addIdentity(GitMonitoringService.this.privateKeyPath, GitMonitoringService.this.passphrase);
```

Comment:
Why is the path case only passing the private key, but the other case is passing both public and private? Shouldn't both cases be the same or if only private key is required then the second case should only pass the private key?

Reply:
With private key path, Jsch by default looks for the corresponding .pub file in the same dir as the private key and hence it is not needed in one of the APIs.